app/src/schemas/toolSchemas.ts

```diff
@@ -77,11 +78,62 @@ export const toolSchema = z
  * The type of a tool definition
  * @see https://platform.openai.com/docs/guides/structured-outputs/supported-schemas
  */
-export type ToolDefinition = z.infer<typeof toolSchema>;
+export type OpenAIToolDefinition = z.infer<typeof openAIToolSchema>;
 
 /**
  * The JSON schema for a tool definition
  */
-export const toolJSONSchema = zodToJsonSchema(toolSchema, {
+export const openAIToolJSONSchema = zodToJsonSchema(openAIToolSchema, {
   removeAdditionalStrategy: "passthrough",
 });
+
+/**
+ * The schema for an OpenAI tool call, this is what a message that calls a tool looks like
+ *
+ * Note: The nested passThrough's are used to allow for extra keys in JSON schema, however, they do not actually
+ * allow for extra keys when the zod schema is used for parsing. This is to allow more flexibility for users
+ * to define their own tool calls according
+ */
+export const openAIToolCallSchema = z.object({
```

Comment:
see comment below, trying to nme things more honestly in prep for tool conversion between providers @cephalization